libraries/ESP8266WiFiMesh/src/TimeTracker.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2019 Anders Löfgren
+ *
+ * License (MIT license):
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "TimeTracker.h"
+#include <Arduino.h>
+
+TimeTracker::TimeTracker(uint32_t creationTimeMs) : _creationTimeMs(creationTimeMs)
```

Comment:
As I said elsewhere, this looks very similar to polledTimeout. Consider migrating to use polledTimeout, or reimplement in terms of it to not reinvent.

Reply:
I've modified ExpiringTimeTracker so it is now based on PolledTimeout, but I found that using PolledTimeout instead of TimeTracker increases heap usage since PolledTimeout is a larger class. During high transmission intensity a large number of TimeTrackers will be stored, so my thinking is that it may be best to keep TimeTracker so it can be used in the cases where  minimal heap usage can make a difference. What do you think?